tests/test_propertyMapPlot.py

```diff
@@ -0,0 +1,356 @@
+        errors = []
+        axes = fig.get_axes()
+
+        # Check the total number of each axis type.
+        totalSkyAxes = sum(isinstance(ax, skyproj.skyaxes.SkyAxes) for ax in axes)
+        totalAxisArtistAxes = sum(isinstance(ax, axisartist.axislines.Axes) for ax in axes)
+        totalColorbarAxes = sum(isinstance(ax, plt.Axes) and ax.get_label() == "<colorbar>" for ax in axes)
+
+        if totalSkyAxes != 3:
+            errors.append(f"Expected 3 SkyAxes but got {totalSkyAxes}.")
+        if totalAxisArtistAxes != 3:
+            errors.append(f"Expected 3 AxisArtist Axes but got {totalAxisArtistAxes}.")
+        if totalColorbarAxes != 3:
+            errors.append(f"Expected 3 colorbar Axes but got {totalColorbarAxes}.")
+
+        # Check histogram axis.
+        self.isHistogramAxis(
+            axes[0],
+            binsCount,
+            ["Full Tract"]
+            + [f"{self.atool.produce.plot.prettyPrintFloat(factor)}x Zoom" for factor in zoomFactors],
+            errors,
+        )
+
+        # Verify x and y labels for histogram.
+        if axes[0].get_xlabel() != xlabel:
+            errors.append(f"Expected x-label '{xlabel}' for histogram but found '{axes[0].get_xlabel()}'.")
+        if axes[0].get_ylabel() != "Normalized Count":
+            errors.append(
+                f"Expected y-label 'Normalized Count' for histogram but found '{axes[0].get_ylabel()}'."
+            )
+
+        self.assertTrue(len(errors) == 0, msg="\n" + "\n".join(errors))
+
+    def validateRGBFractions(self, fig, RGBFraction, rtol=1e-7):
+        """Checks if a matplotlib figure has specified fractions of R, G, and B
+        colors.
+
+        Parameters
+        ----------
+        fig : `~matplotlib.figure.Figure`
+            The figure to check.
+        RGBFraction : `tuple`
+            Tuple containing the desired fractions for red, green, and blue in
+            the image, respectively.
+        rtol : `float`, optional
+            The relative tolerance allowed for the fractions. Default is 1e-7.
+
+        Raises
+        ------
+        AssertionError
+            If the actual fractions of the RGB colors in the image do not match
+            the expected fractions within the given tolerance.
+        """
+
+        # Unpack the desired fractions.
+        rFraction, gFraction, bFraction = RGBFraction
+
+        # Draw the figure so the renderer can grab the pixel buffer.
+        fig.canvas.draw()
+
+        # Convert figure to data array.
+        data = np.array(fig.canvas.renderer.buffer_rgba())[:, :, :3] / 255.0
+
+        # Calculate fractions.
+        rActualFraction = np.sum(data[:, :, 0]) / data.size
+        gActualFraction = np.sum(data[:, :, 1]) / data.size
+        bActualFraction = np.sum(data[:, :, 2]) / data.size
+
+        # Check if the actual fractions meet the expected fractions within the
+        # given tolerance.
+        errors = []
+        if not np.abs(rActualFraction - rFraction) <= rtol:
```

Comment:
This is fine, but we usually use `self.assertFloatsAlmostEqual` which allows an rtol and/or an atol.  I suppose the advantage here is that you get more information on all the channels rather than the first failure, but I wanted to point out the standard tools.

Reply:
Thanks for the heads-up. Correct, I went this route because seeing info on all errors is quite handy for quick debugging, especially when multiple criteria are being checked in one go.